packages/liveblocks-react-ui/src/utils/use-event-source.ts

```diff
@@ -0,0 +1,17 @@
+import type { EventSource } from "@liveblocks/core";
+import { useEffect } from "react";
+
+import { useLatest } from "./use-latest";
+
+export function useEventSource<T>(
+  eventSource: EventSource<T>,
+  callback: () => void
+) {
+  const latestCallback = useLatest(callback);
+
+  useEffect(() => {
+    const unsubscribe = eventSource.subscribe(() => latestCallback.current());
+
+    return unsubscribe;
+  }, [eventSource, latestCallback]);
+}
```

Comment:
Nice! I like this generic hook!

One suggestion here is to only have it rely on the `Observable` API, not the full `EventSource` API, which should ideally be an implementation detail used by the source of the events — i.e. where those events get emitted. Any party interested in listening to those events, are better off just relying on the `Observable` API.

```suggestion
export function useObservable<T>(
  observable: Observable<T>,
  callback: () => void
) {
  const latestCallback = useLatest(callback);

  useEffect(() => {
    const unsubscribe = observable.subscribe(() => latestCallback.current());

    return unsubscribe;
  }, [observable, latestCallback]);
}
```

Reply:
Makes sense! Done [here](https://github.com/liveblocks/liveblocks/pull/2076/commits/72ff05001f3c767a2b373430f0770df0475e44a1).